internal/certdb/validation.go

```diff
@@ -0,0 +1,56 @@
+package certdb
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+)
+
+// ValidateCertificateRequest validates the given CSR string to the following:
+// The string must be a valid PEM string, and should be of type CERTIFICATE REQUEST
+// The PEM string should be able to be parsed into a x509 Certificate Request
+func ValidateCertificateRequest(csrString string) error {
```

Comment:
Do we need the type in the variable name? Ain't `csr` sufficient?

Reply:
yes, I reworked these a bit now.